### DISCUSSION TITLE
Add "Clown" category to motion detectors

### DISCUSSION THREAD

**Discussion by Kafkatrap** [FakeFishGames/Barotrauma#8165] [Ideas and suggestions]
As stated in description, currently motion detectors can differentiate between monsters and humans and such. 

Motion detectors should also be able to be set to "clown" so it only picks up on people wearing the clown masks or clothes. 

Would have a lot of potential when it comes to sub building and RP. 
Doors to Armory simply locking up at the sight of a clown, so they have to force it open with a crowbar or bring a human hostage. Etc, you get the idea.

**Comment by MaaiG** [comment]
> so they have to force it open with a crowbar or bring a human hostage. Etc, you get the idea.

Nice! 🤡

The topic name may be broader... IMHO.
Special rooms = Special security measures!
May be a need to be created a new detectors?
Like a `Retinal scanner`, `Bio scaner` or a `Facial recognition cameras`... kind of like that?

PS In my opinion, a `motion detector` goes beyond its functions when it determines the type of object! It’s a function of other type of detectors.

**Reply by Kafkatrap** [reply]
Yeah that would be a cool alternative to the ID-tags, retinal scanner would be a fail proof system whereas ID-tags can be stolen. 

Well almost fail proof... 
If they were to add a special "capture hostage" animation when grabbing someone with a one handed weapon equipped so you could capture someone at gun point and lead them around, say, to the retinal scanner would make it even better. 

Currently its very slow and annoying when dragging someone, this would be a faster alternative to lead enemies where you want them to go. 

That could also be applicable in scenarios where you manage to grab an outlaw or pirate. Holding 
one of them at gun point would make the outlaws or pirates hesitant to shoot you until you fire the first shot. 

(Also adding a non-gunpoint grab so you can lead people to your sub without the NPC struggling against you, like when rescuing people from a bandit outpost... Right now its a chore)



However this is all waaay too much to ask for, even just a retinal scanner. 

So i was thinking, what about just a clown tag for the motion detector? That wouldn't require that much work i think. Meaning we could all get our hands on that functionality quicker.

**Reply by MaaiG** [reply]
1.
> Yeah that would be a cool alternative to the ID-tags, retinal scanner would be a fail proof system whereas ID-tags can be stolen.

I wouldn't say that just yet. 😇)))

https://user-images.githubusercontent.com/93348499/152225472-596f6209-1649-473e-963f-6fea45525096.mp4

2.
> If they were to add a special "capture hostage" animation

Let them add it!
Paragraph 1 should not be the only option )))

3.
> However this is all waaay too much to ask for, even just a retinal scanner.

Let them add it! )) Why not! ))
Assign different identifier types to different detectors!
Example:
* Motion detector - Motion, Presence.
* Bio scanner - Human, Non-human, Living, Dead, Biological, Unknown.
* Retinal scanner - Retinal identification (ID without card but must be entered in the database).
* Facial recognition cameras - Identification by appearance (maybe mask or impossible - face covered).
* ID card - access card identification.
* Etc.
